jena-core/src/main/java/org/apache/jena/util/PrefixMappingUtils.java

```diff
@@ -0,0 +1,361 @@
+            { throw new UnsupportedOperationException() ; }
+            
+            @Override
+            public PrefixMapping getPrefixMapping() {
+                return prefixMapping ;
+            }
+        } ;
+        return graph2 ;
+    }
+    
+    /**
+     * Analyse the graph to see which prefixes of the graph are in use.
+     * <p>
+     * In the case of overlapping prefixes (where one prefix declaration is has an initial
+     * URI string which imatches another prefix declaration), all are included, though
+     * they may not be used when printing (that depends on the output process). In effect,
+     * this process has "false positives".
+     * <p>
+     * This function does not calculate new prefixes.
+     * 
+     * @see #calcInUsePrefixMappingTTL(Graph)
+     */
+    public static PrefixMapping calcInUsePrefixMapping(Graph graph) {
+        PrefixMapping prefixMapping = graph.getPrefixMapping() ;
+        if ( prefixMapping == null )
+            return null ;
+        return calcInUsePrefixMapping(graph, prefixMapping) ;
+    }
+    
+    /**
+     * Analyse the graph to see which prefixes of the given {@link PrefixMapping} are in
+     * use.
+     * <p>
+     * In the case of overlapping prefixes (where one prefix declaration is has an initial
+     * URI string which imatches another prefix declaration), all are included, though
+     * they may not be used when printing (that depends on the output process). In effect,
+     * this process has "false positives".
+     * <p>
+     * This function does not calculate new prefixes.
+     * 
+     * @see #calcInUsePrefixMappingTTL(Graph, PrefixMapping)
+     */
+    public static PrefixMapping calcInUsePrefixMapping(Graph graph, PrefixMapping prefixMapping) {
+        
+        /* Method:
+         * 
+         * For each URI in the data, look it up in the trie.
+         * after the last ":" for URNs, then see if that is a declared prefix.
+         * 
+         * Exit early if every prefix is accounted for. 
+         */
+       
+        // Map prefix to URI.
+        Map<String, String> pmap = prefixMapping.getNsPrefixMap() ;
+        
+        // Map URI to prefix, with partial lookup (all uri keys that partly match the URI) 
+        Trie<String> trie = new Trie<>() ; 
+        // Chnage this to "add(uri, uri)" to calculate the uris. 
+        pmap.forEach((prefix,uri)-> trie.add(uri, prefix)) ;
+        Iterator<Triple> iter = graph.find(null, null, null) ;
+        // Prefixes in use.
+        // (URIs if "add(uri, uri)")
+        Set<String> inUse = new HashSet<>() ;
+        
+        // Process to apply to each node
+        Consumer<Node> process = (node)->{
+            if ( ! node.isURI() )
+                return ;
+            String uri = node.getURI() ;
+            // Get all prefixes whose URIs are candidates 
+            List<String> hits = trie.partialSearch(uri) ;
+            if ( hits == null || hits.isEmpty() )
```

Comment:
I think `partialSearch` never returns `null`, but harmless to leave the check here.